tests/framework/lib/framework.c

```diff
@@ -150,6 +157,36 @@ mtf_main(int argc, char **argv, struct mtf_test_coll_info *tci)
             argv_home,
             strerror(errno));
         return EX_OSERR;
+    } else {
+        /* If home wasn't provided, use a temporary directory. */
+
+        argv_home = getenv("HSE_TEST_RUNNER_DIR");
+        if (argv_home)
+            goto touch;
+        argv_home = getenv("MESON_BUILD_ROOT");
+        if (argv_home)
+            goto touch;
+        argv_home = "/tmp";
```

Comment:
Is it possible somewhere to pass in the null string for argv_home, in which case (strlen() - 1) will go awry..

Reply:
argv_home is guaranteed to be non-null by the time it gets to the strlen func. See the if above this diff.

Reply:
not non-null, empty string, as in """"

Reply:
Oh whoops. Nice catch. Let me account for that.